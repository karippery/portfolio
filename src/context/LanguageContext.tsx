// src/context/LanguageContext.tsx
import { createContext, ReactNode, useContext, useState } from 'react';

type LanguageContextType = {
  language: 'en' | 'de';
  setLanguage: (lang: 'en' | 'de') => void;
};

const LanguageContext = createContext<LanguageContextType | undefined>(undefined);

type LanguageProviderProps = {
  children: ReactNode;
};

export const LanguageProvider = ({ children }: LanguageProviderProps) => {
  const [language, setLanguage] = useState<'en' | 'de'>('en');
  
  return (
    <LanguageContext.Provider value={{ language, setLanguage }}>
      {children}
    </LanguageContext.Provider>
  );
};

export const useLanguage = () => {
  const context = useContext(LanguageContext);
  if (context === undefined) {
    throw new Error('useLanguage must be used within a LanguageProvider');
  }
  return context;
};