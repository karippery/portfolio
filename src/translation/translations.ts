// src/translations.ts
export type TranslationKeys = {
    // Navigation
    home: string;
    about: string;
    projects: string;
    contact: string;
    description: string;
    
    // Hero Section
    welcome: string;
    role: string;
    experience: string;
    location: string;
    hobbies: string;
    
    // About Section
    aboutTitle: string;
    aboutContent: string;
    techFocusTitle: string;
    backend: string;
    frontend: string;
    devops: string;
    databases: string;
    testing: string;
    
    // Projects Section
    projectsTitle: string;
    viewCode: string;
    viewDemo: string;
    technologies: string;
    
    // Contact Section
    contactTitle: string;
    contactSubtitle: string;
    formName: string;
    formEmail: string;
    formSubject: string;
    formMessage: string;
    formSubmit: string;
    
    // Footer
    copyright: string;
    rights: string;
  };
  
  export const translations: Record<'en' | 'de', TranslationKeys> = {
    en: {
      // Navigation
      home: "Home",
      about: "Who I Am",
      projects: "My Work",
      contact: "Say Hello",
      description: "I’m a full-stack developer with a passion for building web applications",
      
      // Hero Section
      welcome: "Hi, I'm John Raphy",
      role: "Full-Stack Developer",
      experience: "6+ years building digital solutions",
      location: "Based in Braunschweig, Germany",
      hobbies: "Photography | Gaming | Travel",
      
      // About Section
      aboutTitle: "About Me",
      aboutContent: "I specialize in creating robust backend systems and modern web applications with clean, maintainable code. My approach combines technical excellence with practical problem-solving.",
      techFocusTitle: "Technical Focus",
      backend: "Backend Architecture",
      frontend: "Frontend Development",
      devops: "DevOps & CI/CD Pipelines",
      databases: "Database Systems",
      testing: "Testing Frameworks",
      
      // Projects Section
      projectsTitle: "My Projects",
      viewCode: "View Code",
      viewDemo: "View Demo",
      technologies: "Technologies Used",
      
      // Contact Section
      contactTitle: "Get In Touch",
      contactSubtitle: "Have a project in mind? Let's talk!",
      formName: "Your Name",
      formEmail: "Your Email",
      formSubject: "Subject",
      formMessage: "Your Message",
      formSubmit: "Send Message",
      
      // Footer
      copyright: "© {year} John Raphy Karippery",
      rights: "All rights reserved"
    },
    de: {
      // Navigation
      home: "Startseite",
      about: "Über Mich",
      projects: "Projekte",
      contact: "Kontakt",
      description: "Ich bin ein Full-Stack-Entwickler mit einer Leidenschaft für den Bau von Webanwendungen",

      // Hero Section
      welcome: "Hallo, ich bin John Raphy",
      role: "Full-Stack Entwickler",
      experience: "6+ Jahre Erfahrung in der Softwareentwicklung",
      location: "Wohnhaft in Braunschweig, Deutschland",
      hobbies: "Fotografie | Gaming | Reisen",
      
      // About Section
      aboutTitle: "Über Mich",
      aboutContent: "Ich spezialisiere mich auf die Entwicklung robuster Backend-Systeme und moderner Webanwendungen mit sauberem, wartbarem Code. Mein Ansatz verbindet technische Exzellenz mit praktischer Problemlösung.",
      techFocusTitle: "Technische Schwerpunkte",
      backend: "Backend-Architektur",
      frontend: "Frontend-Entwicklung",
      devops: "DevOps & CI/CD Pipelines",
      databases: "Datenbanksysteme",
      testing: "Testframeworks",
      
      // Projects Section
      projectsTitle: "Meine Projekte",
      viewCode: "Code ansehen",
      viewDemo: "Demo ansehen",
      technologies: "Verwendete Technologien",
      
      // Contact Section
      contactTitle: "Kontakt",
      contactSubtitle: "Haben Sie ein Projekt? Lassen Sie uns reden!",
      formName: "Ihr Name",
      formEmail: "Ihre E-Mail",
      formSubject: "Betreff",
      formMessage: "Ihre Nachricht",
      formSubmit: "Nachricht senden",
      
      // Footer
      copyright: "© {year} John Raphy Karippery",
      rights: "Alle Rechte vorbehalten"
    }
  };
  
  // Helper function for dynamic content
  export const translateWithValues = (text: string, values: Record<string, string>) => {
    return Object.entries(values).reduce(
      (result, [key, value]) => result.replace(`{${key}}`, value),
      text
    );
  };