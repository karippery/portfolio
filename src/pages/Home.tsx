import Skills from '../components/Skills';
import { useLanguage } from '../context/LanguageContext';
import Profileimage from '../images/profile.png';
import { translations } from '../translation/translations';

const Home = () => {
  const { language } = useLanguage();
  const t = translations[language];
  return (
    <main className="container mx-auto px-4 py-8"> 
      <section className="mb-12">
        <div className="flex flex-col md:flex-row items-center">
          <img 
            src={Profileimage}
            alt="John Raphy Karippery" 
            className="rounded-full w-40 h-40 object-cover mb-5 md:mb-0 md:mr-9 hover:scale-150 transition-transform duration-300"
          />
          <div className="space-y-6">
            {/* Name/Title Section */}
            <div className="space-y-2">
              <h1 className="text-4xl md:text-5xl font-bold text-gray-900">
                {t.welcome}
              </h1>
            </div>

            {/* Professional Identity */}
            <div className="space-y-4">
              <h2 className="text-2xl md:text-3xl text-gray-700 font-medium">
                {t.description}
              </h2>
              
              <p className="text-lg text-gray-700 leading-relaxed max-w-5xl">
                {t.shortProfile}
                <br />
              </p>
            </div>

            {/* Personal Details */}
            <div className="flex flex-wrap gap-4 pt-2">
                <a 
                  href="https://www.google.com/maps?q=Braunschweig,+Germany" 
                  target="_blank" 
                  rel="noopener noreferrer"
                  className="flex items-center bg-gray-100 px-4 py-2 rounded-full hover:scale-110 transition-transform duration-300"
                >
                  <span className="text-gray-700">📍 {t.location}</span>
                </a>
              <div className="flex items-center bg-gray-100 px-4 py-2 rounded-full hover:scale-110 transition-transform duration-300">
                <span className="text-gray-700">📸 {t.hobbies.photography}</span>
              </div>
              <div className="flex items-center bg-gray-100 px-4 py-2 rounded-full hover:scale-110 transition-transform duration-300">
                <span className="text-gray-700">🌍 {t.hobbies.travel}</span>
              </div>
              <div className="flex items-center bg-gray-100 px-4 py-2 rounded-full hover:scale-110 transition-transform duration-300">
                <span className="text-gray-700">🎮 {t.hobbies.gaming}</span>
              </div>
            </div>
          </div>
        </div>
      </section>
      <section id="skills" className="py-16 px-4 sm:px-6 lg:px-8 bg-gradient-to-b from-gray-50 to-white">
        <Skills />
      </section>
    </main>
  );
};

export default Home;