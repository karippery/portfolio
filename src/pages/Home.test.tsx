// __tests__/Home.test.tsx
import { render, screen } from '@testing-library/react';
import Home from '../pages/Home';
import { LanguageProvider } from '../context/LanguageContext';
import { translations } from '../translation/translations';

// Mock the Skills component
jest.mock('../components/Skills', () => () => <div data-testid="skills-component">Skills Component</div>);

// Mock the image import
jest.mock('../images/profile.png', () => 'profile-image-mock-url');

describe('Home Component', () => {
  const renderHome = (language: 'en' | 'de' = 'en') => {
    return render(
      <LanguageProvider defaultLanguage={language}>
        <Home />
      </LanguageProvider>
    );
  };

  test('renders without crashing', () => {
    renderHome();
    expect(screen.getByRole('main')).toBeInTheDocument();
  });

  test('displays profile image with correct alt text', () => {
    renderHome();
    const img = screen.getByAltText('John Raphy Karippery');
    expect(img).toBeInTheDocument();
    expect(img).toHaveAttribute('src', 'profile-image-mock-url');
    expect(img).toHaveClass('rounded-full');
  });

  describe('English Content', () => {
    beforeEach(() => {
      renderHome('en');
    });

    test('renders welcome message', () => {
      expect(screen.getByText(translations.en.welcome)).toBeInTheDocument();
    });

    test('renders professional description', () => {
      expect(screen.getByText(translations.en.description)).toBeInTheDocument();
    });

    test('renders short profile text', () => {
      expect(screen.getByText(translations.en.shortProfile)).toBeInTheDocument();
    });

    test('renders personal details tags', () => {
      expect(screen.getByText(`📍 ${translations.en.location}`)).toBeInTheDocument();
      expect(screen.getByText(`📸 ${translations.en.hobbies.photography}`)).toBeInTheDocument();
      expect(screen.getByText(`🌍 ${translations.en.hobbies.travel}`)).toBeInTheDocument();
      expect(screen.getByText(`🎮 ${translations.en.hobbies.gaming}`)).toBeInTheDocument();
    });
  });

  describe('German Content', () => {
    beforeEach(() => {
      jest.clearAllMocks();
      renderHome('de');
    });

    test('matches snapshot for English', () => {
      const { asFragment } = renderHome('en');
      expect(asFragment()).toMatchSnapshot();
    });

    test('matches snapshot for German', () => {
      const { asFragment } = renderHome('de');
      expect(asFragment()).toMatchSnapshot();
    });
  });
});