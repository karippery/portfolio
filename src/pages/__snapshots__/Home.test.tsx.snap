// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component German Content matches snapshot for English 1`] = `
<DocumentFragment>
  <main
    class="container mx-auto px-4 py-8"
  >
    <section
      class="mb-12"
    >
      <div
        class="flex flex-col md:flex-row items-center"
      >
        <img
          alt="John Raphy Karippery"
          class="rounded-full w-40 h-40 object-cover mb-5 md:mb-0 md:mr-9 hover:scale-150 transition-transform duration-300"
          src="profile-image-mock-url"
        />
        <div
          class="space-y-6"
        >
          <div
            class="space-y-2"
          >
            <h1
              class="text-4xl md:text-5xl font-bold text-gray-900"
            >
              Hi, I'm John Raphy
            </h1>
          </div>
          <div
            class="space-y-4"
          >
            <h2
              class="text-2xl md:text-3xl text-gray-700 font-medium"
            >
              I’m a software developer with a passion for building web applications
            </h2>
            <p
              class="text-lg text-gray-700 leading-relaxed max-w-5xl"
            >
              For the past 6 years, I've been crafting backend architecturesand web applicationsthat solve real problems. When I'm not coding, you'll find me exploring new technologies or contributing to open-sourceprojects. I thrive in collaborative environments and love sharing knowledge with others. My goal is to create impactful software that enhances user experiences and drives innovation.
              <br />
            </p>
          </div>
          <div
            class="flex flex-wrap gap-4 pt-2"
          >
            <a
              class="flex items-center bg-gray-100 px-4 py-2 rounded-full hover:scale-110 transition-transform duration-300"
              href="https://www.google.com/maps?q=Braunschweig,+Germany"
              rel="noopener noreferrer"
              target="_blank"
            >
              <span
                class="text-gray-700"
              >
                📍 Based in Braunschweig, Germany
              </span>
            </a>
            <div
              class="flex items-center bg-gray-100 px-4 py-2 rounded-full hover:scale-110 transition-transform duration-300"
            >
              <span
                class="text-gray-700"
              >
                📸 Photography
              </span>
            </div>
            <div
              class="flex items-center bg-gray-100 px-4 py-2 rounded-full hover:scale-110 transition-transform duration-300"
            >
              <span
                class="text-gray-700"
              >
                🌍 Traveling
              </span>
            </div>
            <div
              class="flex items-center bg-gray-100 px-4 py-2 rounded-full hover:scale-110 transition-transform duration-300"
            >
              <span
                class="text-gray-700"
              >
                🎮 Gaming
              </span>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      class="py-16 px-4 sm:px-6 lg:px-8 bg-gradient-to-b from-gray-50 to-white"
      id="skills"
    >
      <div
        data-testid="skills-component"
      >
        Skills Component
      </div>
    </section>
  </main>
</DocumentFragment>
`;

exports[`Home Component German Content matches snapshot for German 1`] = `
<DocumentFragment>
  <main
    class="container mx-auto px-4 py-8"
  >
    <section
      class="mb-12"
    >
      <div
        class="flex flex-col md:flex-row items-center"
      >
        <img
          alt="John Raphy Karippery"
          class="rounded-full w-40 h-40 object-cover mb-5 md:mb-0 md:mr-9 hover:scale-150 transition-transform duration-300"
          src="profile-image-mock-url"
        />
        <div
          class="space-y-6"
        >
          <div
            class="space-y-2"
          >
            <h1
              class="text-4xl md:text-5xl font-bold text-gray-900"
            >
              Hi, I'm John Raphy
            </h1>
          </div>
          <div
            class="space-y-4"
          >
            <h2
              class="text-2xl md:text-3xl text-gray-700 font-medium"
            >
              I’m a software developer with a passion for building web applications
            </h2>
            <p
              class="text-lg text-gray-700 leading-relaxed max-w-5xl"
            >
              For the past 6 years, I've been crafting backend architecturesand web applicationsthat solve real problems. When I'm not coding, you'll find me exploring new technologies or contributing to open-sourceprojects. I thrive in collaborative environments and love sharing knowledge with others. My goal is to create impactful software that enhances user experiences and drives innovation.
              <br />
            </p>
          </div>
          <div
            class="flex flex-wrap gap-4 pt-2"
          >
            <a
              class="flex items-center bg-gray-100 px-4 py-2 rounded-full hover:scale-110 transition-transform duration-300"
              href="https://www.google.com/maps?q=Braunschweig,+Germany"
              rel="noopener noreferrer"
              target="_blank"
            >
              <span
                class="text-gray-700"
              >
                📍 Based in Braunschweig, Germany
              </span>
            </a>
            <div
              class="flex items-center bg-gray-100 px-4 py-2 rounded-full hover:scale-110 transition-transform duration-300"
            >
              <span
                class="text-gray-700"
              >
                📸 Photography
              </span>
            </div>
            <div
              class="flex items-center bg-gray-100 px-4 py-2 rounded-full hover:scale-110 transition-transform duration-300"
            >
              <span
                class="text-gray-700"
              >
                🌍 Traveling
              </span>
            </div>
            <div
              class="flex items-center bg-gray-100 px-4 py-2 rounded-full hover:scale-110 transition-transform duration-300"
            >
              <span
                class="text-gray-700"
              >
                🎮 Gaming
              </span>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      class="py-16 px-4 sm:px-6 lg:px-8 bg-gradient-to-b from-gray-50 to-white"
      id="skills"
    >
      <div
        data-testid="skills-component"
      >
        Skills Component
      </div>
    </section>
  </main>
</DocumentFragment>
`;

exports[`Home Component matches snapshot for English 1`] = `
<DocumentFragment>
  <main
    class="container mx-auto px-4 py-8"
  >
    <section
      class="mb-12"
    >
      <div
        class="flex flex-col md:flex-row items-center"
      >
        <img
          alt="John Raphy Karippery"
          class="rounded-full w-40 h-40 object-cover mb-5 md:mb-0 md:mr-9 hover:scale-150 transition-transform duration-300"
          src="profile-image-mock-url"
        />
        <div
          class="space-y-6"
        >
          <div
            class="space-y-2"
          >
            <h1
              class="text-4xl md:text-5xl font-bold text-gray-900"
            >
              Hi, I'm John Raphy
            </h1>
          </div>
          <div
            class="space-y-4"
          >
            <h2
              class="text-2xl md:text-3xl text-gray-700 font-medium"
            >
              I’m a software developer with a passion for building web applications
            </h2>
            <p
              class="text-lg text-gray-700 leading-relaxed max-w-5xl"
            >
              For the past 6 years, I've been crafting backend architecturesand web applicationsthat solve real problems. When I'm not coding, you'll find me exploring new technologies or contributing to open-sourceprojects. I thrive in collaborative environments and love sharing knowledge with others. My goal is to create impactful software that enhances user experiences and drives innovation.
              <br />
            </p>
          </div>
          <div
            class="flex flex-wrap gap-4 pt-2"
          >
            <a
              class="flex items-center bg-gray-100 px-4 py-2 rounded-full hover:scale-110 transition-transform duration-300"
              href="https://www.google.com/maps?q=Braunschweig,+Germany"
              rel="noopener noreferrer"
              target="_blank"
            >
              <span
                class="text-gray-700"
              >
                📍 Based in Braunschweig, Germany
              </span>
            </a>
            <div
              class="flex items-center bg-gray-100 px-4 py-2 rounded-full hover:scale-110 transition-transform duration-300"
            >
              <span
                class="text-gray-700"
              >
                📸 Photography
              </span>
            </div>
            <div
              class="flex items-center bg-gray-100 px-4 py-2 rounded-full hover:scale-110 transition-transform duration-300"
            >
              <span
                class="text-gray-700"
              >
                🌍 Traveling
              </span>
            </div>
            <div
              class="flex items-center bg-gray-100 px-4 py-2 rounded-full hover:scale-110 transition-transform duration-300"
            >
              <span
                class="text-gray-700"
              >
                🎮 Gaming
              </span>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      class="py-16 px-4 sm:px-6 lg:px-8 bg-gradient-to-b from-gray-50 to-white"
      id="skills"
    >
      <div
        data-testid="skills-component"
      >
        Skills Component
      </div>
    </section>
  </main>
</DocumentFragment>
`;

exports[`Home Component matches snapshot for German 1`] = `
<DocumentFragment>
  <main
    class="container mx-auto px-4 py-8"
  >
    <section
      class="mb-12"
    >
      <div
        class="flex flex-col md:flex-row items-center"
      >
        <img
          alt="John Raphy Karippery"
          class="rounded-full w-40 h-40 object-cover mb-5 md:mb-0 md:mr-9 hover:scale-150 transition-transform duration-300"
          src="profile-image-mock-url"
        />
        <div
          class="space-y-6"
        >
          <div
            class="space-y-2"
          >
            <h1
              class="text-4xl md:text-5xl font-bold text-gray-900"
            >
              Hi, I'm John Raphy
            </h1>
          </div>
          <div
            class="space-y-4"
          >
            <h2
              class="text-2xl md:text-3xl text-gray-700 font-medium"
            >
              I’m a software developer with a passion for building web applications
            </h2>
            <p
              class="text-lg text-gray-700 leading-relaxed max-w-5xl"
            >
              For the past 6 years, I've been crafting backend architecturesand web applicationsthat solve real problems. When I'm not coding, you'll find me exploring new technologies or contributing to open-sourceprojects. I thrive in collaborative environments and love sharing knowledge with others. My goal is to create impactful software that enhances user experiences and drives innovation.
              <br />
            </p>
          </div>
          <div
            class="flex flex-wrap gap-4 pt-2"
          >
            <a
              class="flex items-center bg-gray-100 px-4 py-2 rounded-full hover:scale-110 transition-transform duration-300"
              href="https://www.google.com/maps?q=Braunschweig,+Germany"
              rel="noopener noreferrer"
              target="_blank"
            >
              <span
                class="text-gray-700"
              >
                📍 Based in Braunschweig, Germany
              </span>
            </a>
            <div
              class="flex items-center bg-gray-100 px-4 py-2 rounded-full hover:scale-110 transition-transform duration-300"
            >
              <span
                class="text-gray-700"
              >
                📸 Photography
              </span>
            </div>
            <div
              class="flex items-center bg-gray-100 px-4 py-2 rounded-full hover:scale-110 transition-transform duration-300"
            >
              <span
                class="text-gray-700"
              >
                🌍 Traveling
              </span>
            </div>
            <div
              class="flex items-center bg-gray-100 px-4 py-2 rounded-full hover:scale-110 transition-transform duration-300"
            >
              <span
                class="text-gray-700"
              >
                🎮 Gaming
              </span>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      class="py-16 px-4 sm:px-6 lg:px-8 bg-gradient-to-b from-gray-50 to-white"
      id="skills"
    >
      <div
        data-testid="skills-component"
      >
        Skills Component
      </div>
    </section>
  </main>
</DocumentFragment>
`;
